src/natpmp.cpp

```diff
@@ -170,18 +172,22 @@ bool natpmp::get_mapping(int index, int& local_port, int& external_port, int& pr
 }
 
 #ifndef TORRENT_DISABLE_LOGGING
+bool natpmp::should_log() const
+{
+	return m_callback.should_log_portmap(aux::portmap_transport::natpmp);
+}
+
 TORRENT_FORMAT(2, 3)
 void natpmp::log(char const* fmt, ...) const
 {
 	TORRENT_ASSERT(is_single_thread());
+	if (!should_log()) return;
```

Comment:
I don't think so, I don't check in every `log`, just in the expensive ones, however, if a call is called (say with a single string) I want to avoid the `va_*` and `snprinf`.